testdata/p4_16_samples_outputs/issue2201-bmv2-midend.p4

```diff
@@ -137,10 +165,10 @@ control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_m
     @hidden action issue2201bmv2l164() {
         log_msg<tuple_6>("GREPME serenum1={}", (tuple_6){f0 = serenum1_0});
         log_msg<tuple_0>("GREPME hdr.ethernet.isValid()={}", (tuple_0){f0 = hdr.ethernet.isValid()});
-        log_msg<tuple_7>("GREPME hdr.ethernet={}", (tuple_7){f0 = hdr.ethernet});
+        log_msg<tuple_7>("GREPME hdr.ethernet=(dstAddr:{},srcAddr:{},etherType:{})", (tuple_7){f0 = hdr.ethernet.dstAddr,f1 = hdr.ethernet.srcAddr,f2 = hdr.ethernet.etherType});
```

Comment:
Is `dstAddr:{}` expected here?

Reply:
I believe the `{}` part is like Python's "string containing one or more {} sub-strings".format(one or more values to format), i.e. it gets replaced by the corresponding value from the tuple argument.

Reply:
@jhavrane Have you tried executing these P4 programs on BMv2 simple_switch to verify that the output looks reasonable, with these changes?

Reply:
@jafingerhut I've run the issue2201-bmv2.p4 on simple_switch with custom pcap and the output looks fine to me. Here is a log for clarity:
```
Calling target program-options parser
[09:33:31.967] [bmv2] [D] [thread 3895149] Set default default entry for table 'tbl_issue2201bmv2l101': issue2201bmv2l101 -
[09:33:31.968] [bmv2] [D] [thread 3895149] Set default default entry for table 'tbl_issue2201bmv2l139': issue2201bmv2l139 -
[09:33:31.968] [bmv2] [D] [thread 3895149] Set default default entry for table 'tbl_issue2201bmv2l141': issue2201bmv2l141 -
[09:33:31.968] [bmv2] [D] [thread 3895149] Set default default entry for table 'tbl_issue2201bmv2l143': issue2201bmv2l143 -
[09:33:31.968] [bmv2] [D] [thread 3895149] Set default default entry for table 'tbl_issue2201bmv2l152': issue2201bmv2l152 -
[09:33:31.968] [bmv2] [D] [thread 3895149] Set default default entry for table 'tbl_issue2201bmv2l156': issue2201bmv2l156 -
[09:33:31.968] [bmv2] [D] [thread 3895149] Set default default entry for table 'tbl_issue2201bmv2l158': issue2201bmv2l158 -
[09:33:31.968] [bmv2] [D] [thread 3895149] Set default default entry for table 'tbl_issue2201bmv2l160': issue2201bmv2l160 -
[09:33:31.968] [bmv2] [D] [thread 3895149] Set default default entry for table 'tbl_issue2201bmv2l164': issue2201bmv2l164 -
Adding interface eth as port 0 (files)
[09:33:31.968] [bmv2] [D] [thread 3895149] Adding interface eth as port 0
[09:33:31.968] [bmv2] [I] [thread 3895149] Starting Thrift server on port 9090
[09:33:31.968] [bmv2] [I] [thread 3895149] Thrift server was started
[09:33:31.968] [bmv2] [D] [thread 3895153] Pcap reader: starting PcapFilesReader 1
[09:33:31.968] [bmv2] [D] [thread 3895153] Pcap reader: waiting for 1 seconds
[09:33:32.968] [bmv2] [D] [thread 3895153] Pcap reader: first packet time 1733208667:748600
[09:33:32.968] [bmv2] [D] [thread 3895153] Pcap reader: first packet to send 1733208667:748600
[09:33:32.969] [bmv2] [D] [thread 3895153] Pcap reader: first packet to send 1733208667:749417
[09:33:32.969] [bmv2] [D] [thread 3895153] Pcap reader: first packet to send 1733208667:795575
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Processing packet received on port 0
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Parser 'parser': start
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Parser 'parser' entering state 'start'
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Extracting header 'ethernet'
[09:33:32.969] [bmv2] [D] [thread 3895153] Pcap reader: first packet to send 1733208667:985508
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Parser state 'start': key is 0800
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Parser 'parser' entering state 'parse_ipv4'
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Extracting header 'ipv4'
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Parser state 'parse_ipv4' has no switch, going to default next state
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Parser 'parser': end
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Pipeline 'ingress': start
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Looking up key:

[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Table 'tbl_issue2201bmv2l101': miss
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Action entry is issue2201bmv2l101 -
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME Packet ingress begin
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME bool1=0
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME (bit<1>) bool1=0
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME (bit<1>) (!bool1)=1
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME bit1=0
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME signed1=-128
[09:33:32.969] [bmv2] [D] [thread 3895153] Pcap reader: first packet to send 1733208667:985934
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME unsigned1=128
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME hdr.ethernet.dstAddr[1:0]=0
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Looking up key:

[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Table 'tbl_issue2201bmv2l139': miss
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Action entry is issue2201bmv2l139 -
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Looking up key:

[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Table 'tbl_issue2201bmv2l152': miss
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Action entry is issue2201bmv2l152 -
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME enum1=0
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Looking up key:

[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Table 'tbl_issue2201bmv2l156': miss
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Action entry is issue2201bmv2l156 -
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Looking up key:

[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Table 'tbl_issue2201bmv2l164': miss
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Action entry is issue2201bmv2l164 -
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME serenum1=17
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME hdr.ethernet.isValid()=1
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME hdr.ethernet=(dstAddr:65978312167196,srcAddr:127083103933239,etherType:2048)
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME hdr.ipv4.isValid()=1
[09:33:32.969] [bmv2] [D] [thread 3895153] Pcap reader: first packet to send 1733208667:985951
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME hdr.ipv4=(version:4,ihl:5,diffserv:72,totalLen:52,identification:39116,flags:2,fragOffset:0,ttl:57,protocol:6,hdrChecksum:30205,srcAddr:113898351,dstAddr:183732013)
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME stdmeta=(ingress_port:0,egress_spec:0,egress_port:0,instance_type:0,packet_length:66,enq_timestamp:0,enq_qdepth:0,deq_timedelta:0,deq_qdepth:0,ingress_global_timestamp:1048696,egress_global_timestamp:0,mcast_grp:0,egress_rid:0,checksum_error:0,parser_error:0,priority:0)
[09:33:32.969] [bmv2] [I] [thread 3895154] GREPME error.PacketTooShort=1
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Pipeline 'ingress': end
[09:33:32.969] [bmv2] [D] [thread 3895154] [0.0] [cxt 0] Egress port is 0
```